[Lutron] Update initialization and device state logic in RA2 device handlers

bobadair opened the pull request:
Hi. This PR contains an improvement to the Lutron binding that updates initialization and device state logic in all RA2 device handlers to match that in the new shade handler, as discussed with @cweitkamp  in PR 3911 comments.

More specifically, it adds the following:
-	When setting device status, take into account bridge status and ability to query device (if queryable).
-	Update appropriate device state whenever bridge status transitions to online from offline.

  Regards,
    Bob

cweitkamp commented on addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/handler/BaseKeypadHandler.java:
```diff
@@ -195,12 +197,49 @@ private void asyncInitialize() {
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
```
Did you think about moving this method to the `LutronHandler` instead of implement it in each handler?

bobadair replied:
Yes, that's what I had intended to do originally.  But there were some complications with putting it there, so I'm afraid I took the lazy way out.  I've now done it properly. :-)

bobadair commented:
Thanks @cweitkamp . I don't think there should be any merge conflicts with #2325, but if there are I should be able to resolve them quickly.  In the meantime, I've pushed another commit that addresses your comment.
I'll be creating another PR sometime in the next day or two for Issue #4050.  But that is independent of this one, and also should not conflict with #2325.

cweitkamp commented on addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/handler/BaseKeypadHandler.java:
```diff
@@ -195,12 +196,39 @@ private void asyncInitialize() {
+                    // To reduce query volume, query only 1st LED and LEDs with linked channels.
+                    for (KeypadComponent component : ledList) {
+                        if (component.id() == ledList.get(0).id() || isLinked(channelFromComponent(component.id()))) {
+                            queryDevice(component.id(), ACTION_LED_STATE);
```
I guess we can break the `for` loop after the first match. Can we?

bobadair replied:
No, we needs to iterate through all of them so that any button LED with a linked channel can be queried. Their states need to be updated when the bridge status transitions back to ONLINE from OFFLINE, since they may be stale.  I query the first one just in case no LED channels are linked, since we need to generate at least one query and get a response to finish updating the device state.

cweitkamp replied:
:thumbsup:

cweitkamp commented on addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/handler/LutronHandler.java:
```diff
@@ -23,7 +24,7 @@
  *
  * @author Allan Tong - Initial contribution
- *
+ * @author Bob Adair - Added methods for status and state management
```
Do you want to add the `@author` tag to every class you have changed in this PR?

bobadair replied:
I guess I should. If only for purposes of allocating blame. Done.

cweitkamp replied:
No, absolutely not. Feel free to ommit them, but do not mix it up.
